patches/0036-Disable-retry-for-KMS-access-denied-in-lambda.patch

```diff
@@ -43,71 +42,42 @@ index 0000000000..4625f9e31c
 +		}
 +		return aws_sdkv2.UnknownTernary // Delegate
 +	})
-+	return lambda_sdkv2.NewFromConfig(cfg, func(o *lambda_sdkv2.Options) {
-+		if endpoint := config[names.AttrEndpoint].(string); endpoint != "" {
-+			tflog.Debug(ctx, "setting endpoint", map[string]any{
-+				"tf_aws.endpoint": endpoint,
-+			})
-+			o.BaseEndpoint = aws_sdkv2.String(endpoint)
 +
-+			if o.EndpointOptions.UseFIPSEndpoint == aws_sdkv2.FIPSEndpointStateEnabled {
-+				tflog.Debug(ctx, "endpoint set, ignoring UseFIPSEndpoint setting")
-+				o.EndpointOptions.UseFIPSEndpoint = aws_sdkv2.FIPSEndpointStateDisabled
-+			}
-+		}
-+		o.Retryer = conns.AddIsErrorRetryables(cfg.Retryer().(aws_sdkv2.RetryerV2), retry)
-+	}), nil
++	return lambda_sdkv2.NewFromConfig(cfg,
```

Comment:
There's multiple similar changes to patches in this PR.
Upstream introduces some helpers for configuring the endpoint: https://github.com/hashicorp/terraform-provider-aws/pull/38057/files